### PR TITLE
chore: swapping types in the main oracle to use ticker

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: aggregator/aggregator_test.go [hunk 1]
```diff
@@ -5,44 +5,42 @@ import (
...
+	ethusd  = mmtypes.NewTicker("ETHEREUM", "USD", 8, 1)
+	usdtusd = mmtypes.NewTicker("USDT", "USD", 8, 1)
 )
 
 func TestComputeMedian(t *testing.T) {
 	testCases := []struct {
 		name           string
-		providerPrices aggregator.AggregatedProviderData[string, map[slinkytypes.CurrencyPair]*big.Int]
-		expectedPrices map[slinkytypes.CurrencyPair]*big.Int
+		providerPrices aggregator.AggregatedProviderData[string, map[mmtypes.Ticker]*big.Int]
```
Comment (aljo242) [inline aggregator/aggregator_test.go:15]:
```suggestion
		providerPrices aggregator.AggregatedProviderPriceData
```

with type alias

File: oracle/metrics_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func (s *OracleMetricsTestSuite) SetupTest() {
 	s.o, err = oracle.New(
 		oracle.WithUpdateInterval(oracleTicker),
 		oracle.WithProviders(
-			[]providertypes.Provider[slinkytypes.CurrencyPair, *big.Int]{
+			[]providertypes.Provider[mmtypes.Ticker, *big.Int]{
```
Comment (aljo242) [inline oracle/metrics_test.go]:
```suggestion
			[]providertypes.PriceProvider{
```

with type alias

File: oracle/options.go [hunk 3]
```diff
@@ -57,20 +57,20 @@ func WithMetricsConfig(config config.MetricsConfig) Option {
 }
 
 // WithAggregateFunction sets the aggregate function on the Oracle.
-func WithAggregateFunction(fn aggregator.AggregateFn[string, map[slinkytypes.CurrencyPair]*big.Int]) Option {
+func WithAggregateFunction(fn aggregator.AggregateFn[string, map[mmtypes.Ticker]*big.Int]) Option {
```
Comment (aljo242) [inline oracle/options.go]: alias

File: oracle/oracle.go [hunk 4]
```diff
@@ -207,7 +207,7 @@ func (o *OracleImpl) fetchPrices(provider providertypes.Provider[slinkytypes.Cur
 			provider.Name(),
 			string(provider.Type()),
 			strings.ToLower(pair.String()),
-			pair.Decimals(),
+			int(pair.Decimals),
```
Comment (aljo242) [inline oracle/oracle.go]: Can we change this metric to take in `uint64`?

File: oracle/oracle.go [hunk 5]
```diff
@@ -270,7 +270,7 @@ func (o *OracleImpl) GetPrices() map[slinkytypes.CurrencyPair]*big.Int {
 
 			o.metrics.UpdateAggregatePrice(
 				strings.ToLower(cp.String()),
-				cp.Decimals(),
+				int(cp.Decimals),
```
Comment (aljo242) [inline oracle/oracle.go]: same here

File: oracle/oracle_test.go [hunk 6]
```diff
@@ -70,33 +70,37 @@ func (s *OracleTestSuite) SetupTest() {
 	s.random = rand.New(rand.NewSource(time.Now().UnixNano()))
 	s.logger = zap.NewExample()
 
-	s.currencyPairs = []slinkytypes.CurrencyPair{
-		slinkytypes.NewCurrencyPair("BITCOIN", "USD"),
-		slinkytypes.NewCurrencyPair("ETHEREUM", "USD"),
-		slinkytypes.NewCurrencyPair("COSMOS", "USD"),
+	btcusd := mmtypes.NewTicker("BITCOIN", "USD", 8, 1)
+	ethusd := mmtypes.NewTicker("ETHEREUM", "USD", 8, 1)
+	atomusd := mmtypes.NewTicker("COSMOS", "USD", 8, 1)
```
Comment (aljo242) [inline oracle/oracle_test.go]: Use constants?

File: oracle/providers_test.go [hunk 7]
```diff
@@ -23,85 +23,85 @@ func (s *OracleTestSuite) TestProviders() {
...
+				resolved := map[mmtypes.Ticker]providertypes.Result[*big.Int]{
 					s.currencyPairs[0]: {
 						Value:     big.NewInt(100),
 						Timestamp: time.Date(9999, 1, 1, 0, 0, 0, 0, time.UTC),
 					},
 				}
-				response := providertypes.NewGetResponse[slinkytypes.CurrencyPair, *big.Int](resolved, nil)
-				responses := []providertypes.GetResponse[slinkytypes.CurrencyPair, *big.Int]{response}
-				provider := testutils.CreateAPIProviderWithGetResponses[slinkytypes.CurrencyPair, *big.Int](
+				response := providertypes.NewGetResponse[mmtypes.Ticker, *big.Int](resolved, nil)
```
Comment (aljo242) [inline oracle/providers_test.go:86]: alias city

**Review by Eric-Warehime — APPROVED** [review]

**Pull request merged**